bin/backport-wp-commit.sh

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env bash
+
+set -e
+
+wp_changeset="$1"
+if [[ ! "$wp_changeset" =~ ^[0-9]+$ ]]; then
+	echo "Usage: $0 WP_CHANGESET_NUMBER [--current-branch]"
+	echo
+	echo "  WP_CHANGESET_NUMBER  The SVN changeset number to port.  If this change depends"
+	echo "                       on other changes, make sure they have already been ported."
+	echo "  --current-branch     Apply the commit directly to the current branch instead of"
+	echo "                       creating a new branch."
+	echo
+	exit 1
+fi
+
+create_branch=yes
+if [ "$2" = "--current-branch" ]; then
+	create_branch=no
+fi
+
+# Sanity check: make sure we have a .git directory, and at least one remote
+# pointing to a repository named ClassicPress
+if [ ! -d .git ] || ! git remote -v | grep -q '\b/ClassicPress\b'; then
+	echo "ERROR: Call this script from within your ClassicPress repository"
+	exit 1
+fi
+
+# Make sure there are no modified files in the local repository
+change_type=
+if ! git diff --exit-code --quiet; then
+	change_type="Modified file(s)"
+elif ! git diff --cached --exit-code --quiet; then
+	change_type="Staged file(s)"
+fi
+if [ ! -z "$change_type" ]; then
+	git status
+	echo
+	echo "ERROR: $change_type detected"
+	echo "ERROR: You must start this script from a clean working tree!"
+	exit 1
+fi
+
+cmd() {
+	echo "+" "$@"
+	"$@" 2>&1 | sed 's/^/> /'
+	return ${PIPESTATUS[0]}
+}
+
+wp_remote=$(git remote -v | grep '\bWordPress/wordpress-develop\b' | awk 'END { print $1 }')
+if [ -z "$wp_remote" ]; then
+	echo "Adding git remote 'wp' for WordPress/wordpress-develop"
+	cmd git remote add wp https://github.com/WordPress/wordpress-develop.git
+	wp_remote="wp"
+else
+	echo "Found git remote '$wp_remote' for WordPress/wordpress-develop"
+fi
+
+cp_remote=$(git remote -v | grep '\bClassicPress/ClassicPress\b' | awk 'END { print $1 }')
+if [ -z "$cp_remote" ]; then
+	echo "Adding git remote 'cp' for ClassicPress/ClassicPress"
+	cmd git remote add cp https://github.com/ClassicPress/ClassicPress.git
+	cp_remote="cp"
+else
+	echo "Found git remote '$cp_remote' for ClassicPress/ClassicPress"
+fi
+
+echo "Updating repositories from GitHub"
+cmd git fetch "$wp_remote"
+cmd git fetch "$cp_remote"
+
+commit_hash=
```

Comment:
I am not sure if it is the case to leave this variable without a value

Reply:
It's the same as `commit_hash=""`, we can change it if that's clearer.

Later on we check `[ -z "$commit_hash" ]` to see if the commit was not found.